pilot/pkg/networking/core/cluster.go

```diff
@@ -85,7 +85,7 @@ func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, upd
 	// Holds subset clusters per service, keyed by hostname.
 	subsetClusters := make(map[string]sets.String)
 
-	for _, cluster := range watched.ResourceNames {
+	for cluster := range watched.ResourceNames {
```

Comment:
By this, it it making the generated clusters(actually all subscribed resources) unordered. This might be a risk, i donot know all the proxies though

Reply:
there is no ordering semantics on discovery response. We already are basically unordered since we do not actually sort the resource names btw - now we are just unordered in a different way